### ISSUE TITLE
Exclude any part from storage key

### ISSUE THREAD

**Issue by olenagerasimova** [artipie/asto#341]
Let's add class to exclude any part (by part string value or index) from the existing key and create a new key without this excluded part. By a part I mean a string between `/` chars, a part cannot contain separator char `\`. 
Let's suppose the required class is called `KeyParts`, here is an example:
```java
// exclude part by part value
new KeyParts(new Key.From("one", "two")).exclude("one") // result is Key instance new Key.From("two")
new KeyParts(new Key.From("one", "two")).exclude("two") // result is Key instance new Key.From("one")
// exclude part by index (starting with 0)
new KeyParts(new Key.From("one", "two")).exclude(0) // result is Key instance new Key.From("two")
new KeyParts(new Key.From("one", "two")).exclude(1) // result is Key instance new Key.From("one")

// `exclude` should exclude all found parts
new KeyParts(new Key.From("one", "two", "one", "three")).exclude("one") // result is Key instance new Key.From("two", "three")
// we can also add `excludeFirst` and `excludeLast` to exclude first/last part
new KeyParts(new Key.From("one", "two", "one", "three")).excludeFirst("one") // result is Key instance new Key.From("two", "one", "three")
```
If part to exclude was not found (or index is out of bounds), the class can return the origin key. Also, this same class can be extended with methods to append the part in the middle of the key, correspondingly to `List.add(object, index)` method, get key parts as list or stream etc.

**Comment by g4s8** [comment]
@olenagerasimova it's really clear what is the "part" here - it's just a substring or a string between `/` chars? Could you clarify and add some examples to help understand and write unit tests?

**Comment by 0pdd** [comment]
@olenagerasimova the puzzle [#352](https://github.com/artipie/asto/issues/352) is still not solved.

**Comment by 0pdd** [comment]
@olenagerasimova 2 puzzles [#352](https://github.com/artipie/asto/issues/352), [#369](https://github.com/artipie/asto/issues/369) are still not solved.

**Comment by 0pdd** [comment]
@olenagerasimova the puzzle [#369](https://github.com/artipie/asto/issues/369) is still not solved; solved: [#352](https://github.com/artipie/asto/issues/352).
